contracts/campaigns/KOLReferral.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@opengsn/contracts/src/ERC2771Recipient.sol";
+
+contract KOLReferral is Initializable, ERC2771Recipient {
+    address public admin;
+
+    mapping(address => string) public walletToReferralId;
+    mapping(string => address) public referralIdToWallet;
+    mapping(address => string) public userReferredBy;
+    mapping(address => address[]) private _kolToUsers;
+
+    address[] private _users;
+    address[] private _kols;
+
+    modifier onlyAdmin() {
+        require(_msgSender() == admin, "Only Admin");
+        _;
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
```

Comment:
Do we need it?

Reply:
Openzepplin suggests to use it.
Halborn guys also recommended during BNBx audit.
https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#initializing_the_implementation_contract